SmartDeviceLink/private/SDLIAPDataSession.m

```diff
@@ -102,6 +104,8 @@ - (void)destroySessionWithCompletionHandler:(void (^)(void))disconnectCompletion
         BOOL cancelledSuccessfully = [strongSelf sdl_isIOThreadCancelled];
         if (!cancelledSuccessfully) {
             SDLLogE(@"The I/O streams were not shut down successfully. We might not be able to create a new session with an accessory during the same app session. If this happens, only force quitting and restarting the app will allow new sessions.");
+        } else {
+            strongSelf.ioStreamThread = nil;
```

Comment:
This is another thing I was looking for but couldn't figure out, the runloop logic is checking for if the ioStreamThread isn't nil but didn't ever see it getting set to nil here, just cancelled. We've been working on an assumption here that this is triggered when there are two connections going on, and the two cause a deadlock. Unsure on the trigger, but if https://github.com/smartdevicelink/sdl_ios/pull/1806/files#diff-f127f5f9119210b4566551d521031b0aL71 were called from different threads I don't think that operation is atomic either which would allow for it if I'm reading it right?

Reply:
Setting to nil makes sense because it might have impact loop's conditions. Also it fixes misleading error messages if "cancelldSuccessfully" is set to NO because the thread is already cancelled hence cannot send the semaphore signal.

I'm not so sure if we have invalidated this assumption in the separate discussion we had. Two connections should  lead to two EA sessions with separate streams and separate threads. A deadlock should not be possible with the SDL architecture. Please let me know if it's still valid.